app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -514,4 +540,27 @@ public void run() {
             }
         });
     }
+
+    public void processStreams(List<EventsBranch> events) {
+        for (EventsBranch event : events) {
+            StreamWrapper streamEvent = (StreamWrapper) event;
+            if (streamEvent.getOperation().equalsIgnoreCase(StreamWrapper.OP_OCCUPY)) {
+                Dao<Stream, Integer> streamDao = app.getDao(Stream.class);
+                List<Stream> streams = streamEvent.getStreams();
+                for (Stream stream : streams) {
+                    try {
+                        // use default color for not subscribed streams
+                        stream.setDefaultColor();
+                        streamDao.createIfNotExists(stream);
+                    } catch (SQLException e) {
+                        ZLog.logException(e);
+                    }
+                }
+            } else {
+                // TODO: handle other operations of stream event
```

Comment:
https://github.com/zulip/zulip/blob/master/static/js/server_events.js#L147
I think we are currently maintaining a list of all streams and subscribed streams through register and events, hence I didn't implement other events.